src/check/constrain/generate/operation.rs

```diff
@@ -159,12 +160,16 @@ pub fn gen_op(
             generate(left, env, ctx, constr)?;
             Ok(env.clone())
         }
-        Node::IsA { left, right } | Node::IsNA { left, right } => {
-            let bool = Expected::new(ast.pos, &Type { name: Name::from(clss::BOOL) });
-            constr.add("and", &Expected::try_from((ast, &env.var_mappings))?, &bool);
-            generate(right, env, ctx, constr)?;
+        Node::IsA { left, right } | Node::IsNA { left, right } => if let Node::Id { .. } = right.node {
```

Comment:
This logic should be implemented in it's own PR.
This changes the logic from `isa` which was an accidental copy past of `is`.
Now, the right must be an identifier identifying a class.